File: StreamCloud/app/src/main/java/kotlin/coroutines/jvm/internal/SpillingKt.java
package kotlin.coroutines.jvm.internal;

/**
 * Backward-compatibility stub for plugins compiled with Kotlin 1.8.x / 1.9.x.
 *
 * Kotlin 2.0 removed SpillingKt from the stdlib. Plugins compiled with older Kotlin
 * reference it via bytecode such as:
 *
 *   getstatic  kotlin/coroutines/jvm/internal/SpillingKt.SPILLING_ENABLED Z
 *   invokestatic kotlin/coroutines/jvm/internal/SpillingKt.nullOutSpilledVariable(Ljava/lang/Object;)Ljava/lang/Object;
 *
 * This Java stub satisfies the class-loader at runtime. Java (unlike Kotlin 2.0's K2
 * compiler) has no restriction against third-party code residing in the kotlin.* package.
 */
@SuppressWarnings({"unused", "SpellCheckingInspection"})
public final class SpillingKt {

    /** Matches the @JvmField static field referenced by Kotlin 1.8/1.9-compiled plugins. */
    public static final boolean SPILLING_ENABLED = true;

    /** Kotlin object companion — some call sites access SpillingKt.INSTANCE. */
    public static final SpillingKt INSTANCE = new SpillingKt();

    private SpillingKt() {}

    /**
     * Inline function from Kotlin 1.9 stdlib — nulls out a spilled coroutine variable.
     * Signature: (Ljava/lang/Object;)Ljava/lang/Object;
     * The original implementation was: inline fun <T> nullOutSpilledVariable(value: T): T = null as T
     */
    @SuppressWarnings("TypeParameterUnusedInFormals")
    public static <T> T nullOutSpilledVariable(T value) {
        return null;
    }
}

File: StreamCloud/app/src/main/java/kotlin/coroutines/jvm/internal/SpillingKt.kt
@file:Suppress("PackageDirectoryMismatch", "PACKAGE_OR_CLASSIFIER_REDECLARATION",
    "NOTHING_TO_INLINE", "unused")
package kotlin.coroutines.jvm.internal

/**
 * Backward-compatibility stub for plugins compiled with Kotlin 1.8.x / 1.9.x.
 *
 * In Kotlin 1.8.20, SpillingKt was added to the JVM coroutines runtime to control
 * whether local variables are "spilled" into coroutine fields on suspension.  Compiled
 * plugins contain bytecode like:
 *
 *   getstatic kotlin/coroutines/jvm/internal/SpillingKt.SPILLING_ENABLED Z
 *   ifeq ...
 *
 * Kotlin 2.0 removed this class, so loading such a plugin on an app compiled with
 * Kotlin 2.0 throws NoClassDefFoundError.  This empty stub — placed at the same
 * package path — satisfies the class-loader and lets the coroutine state machine
 * continue normally.
 *
 * SPILLING_ENABLED = true preserves the original spilling behavior.
 */
@Suppress("unused")
internal object SpillingKt {
    @JvmField
    val SPILLING_ENABLED: Boolean = true
}
